src/dswx_sar/mosaic_rtc_burst.py

```diff
@@ -109,197 +174,127 @@ def check_reprojection(geogrid_mosaic,
-    output_dirname = os.path.dirname(output_tif_path)
-
-    if epsg == epsg_output:
-        
-        dswx_sar_util.save_raster_gdal(data, 
-                    output_tif_path, 
-                    geotransform,
-                    projection, 
-                    scratch_dir=output_dirname,
-                    DataType=dtype)
-    else:
-        output_tif_temp_dir_path = os.path.dirname(output_tif_path)
-        output_tif_temp_base_path = f'{os.path.basename(output_tif_path)}_temp.tif'
-        output_tif_temp_path = os.path.join(output_tif_temp_dir_path, 
-                                            output_tif_temp_base_path)
-        dswx_sar_util.save_raster_gdal(data, 
-            output_tif_temp_path, 
-            geotransform,
-            projection, 
-            scratch_dir=output_tif_temp_dir_path,
-            DataType=dtype)
-
-        opt = gdal.WarpOptions(dstSRS=f'EPSG:{epsg_output}',
-                     xRes=xres, 
-                     yRes=xres,
-                     resampleAlg='nearest',
-                     format='GTIFF')
-        ds = gdal.Warp(output_tif_path, output_tif_temp_path, options=opt)   
-        ds = None 
-
-        os.remove(output_tif_temp_path)
-
-def compute_weighted_mosaic_array(list_rtc_images, list_nlooks,
-                     geogrid_in=None, verbose = True):
+def compute_mosaic_array(list_rtc_images, list_nlooks, mosaic_mode, scratch_dir='',
+                         geogrid_in=None, temp_files_list=None, verbose=True):
     '''
     Mosaic S-1 geobursts and return the mosaic as dictionary
-    paremeters:
+
+    Parameters
     -----------
-        list_rtc: list
-            List of the path to the rtc geobursts
-        list_nlooks: list
-            List of the nlooks raster that corresponds to list_rtc
-        geogrid_in: isce3.product.GeoGridParameters, default: None
+       list_rtc: list
+           List of the path to the rtc geobursts
+       list_nlooks: list
+           List of the nlooks raster that corresponds to list_rtc
+       mosaic_mode: str
+            Mosaic mode. Choices: "average", "first", and "bursts_center"
+       scratch_dir: str (optional)
+            Directory for temporary files
+       geogrid_in: isce3.product.GeoGridParameters, default: None
             Geogrid information to determine the output mosaic's shape and projection
             The geogrid of the output mosaic will automatically determined when it is None
-    Returns:
+       temp_files_list: list (optional)
+            Mutable list of temporary files. If provided,
+            paths to the temporary files generated will be
+            appended to this list
+       verbose: flag (optional)
+            Flag to enable (True) or disable (False) the verbose mode
+    Returns
         mosaic_dict: dict
             Mosaic dictionary
     '''
 
-    num_raster = len(list_rtc_images)
+    mosaic_mode_choices_list = ['average', 'first', 'bursts_center']
```

Comment:
Just a heads up, me and Gustavo are aware of some issue in `bursts_center` mode. You might need to update the related codes later on.